pkg/repository/testworkflow/mongo.go

```diff
@@ -241,8 +274,13 @@ func (r *MongoRepository) GetExecutions(ctx context.Context, filter Filter) (res
 	return
 }
 
+type TestWorkflowExecutionSummaryWithResolvedWorkflow struct {
+	testkube.TestWorkflowExecutionSummary `json:",inline" bson:",inline"`
+	ResolvedWorkflow                      *testkube.TestWorkflow `json:"resolvedWorkflow,omitempty"`
+}
+
 func (r *MongoRepository) GetExecutionsSummary(ctx context.Context, filter Filter) (result []testkube.TestWorkflowExecutionSummary, err error) {
-	result = make([]testkube.TestWorkflowExecutionSummary, 0)
+	executions := make([]TestWorkflowExecutionSummaryWithResolvedWorkflow, 0)
```

Comment:
Better to avoid fetching unnecessary data there. You can either - change the query to fetch only important part, or even better - adjust the aggregate to provide these data.

Reply:
Also, we have different pipelines for Enterprise, so it's very important to test it through.